tests/src/core/testqgsweakrelation.cpp

```diff
@@ -66,72 +67,172 @@ void TestQgsWeakRelation::cleanup()
 
 void TestQgsWeakRelation::testResolved()
 {
-  const QList<QgsRelation::FieldPair> fieldPairs {{ "fk_province", "pk" }};
-
   QgsWeakRelation weakRel( QStringLiteral( "my_relation_id" ),
                            QStringLiteral( "my_relation_name" ),
-                           QgsRelation::RelationStrength::Association,
+                           Qgis::RelationshipStrength::Association,
                            QStringLiteral( "referencingLayerId" ),
                            QStringLiteral( "referencingLayerName" ),
                            QStringLiteral( "Point?crs=epsg:4326&field=pk:int&field=fk_province:int&field=fk_municipality:int" ),
                            QStringLiteral( "memory" ),
                            QStringLiteral( "referencedLayerId" ),
                            QStringLiteral( "referencedLayerName" ),
                            QStringLiteral( "Polygon?crs=epsg:4326&field=pk:int&field=province:int&field=municipality:string" ),
-                           QStringLiteral( "memory" ),
-                           fieldPairs
+                           QStringLiteral( "memory" )
                          );
-  QVERIFY( ! weakRel.resolvedRelation( QgsProject::instance(), QgsVectorLayerRef::MatchType::Name ).isValid() );
+  weakRel.setReferencingLayerFields( {"fk_province" } );
+  weakRel.setReferencedLayerFields( {"pk" } );
```

Comment:
Auto-formatter should pick those up if possible.

```suggestion
  weakRel.setReferencedLayerFields( { "pk" } );
```